boards/arm/nrf52840dk_nrf52840/snippets/mcuboot.overlay

```diff
@@ -0,0 +1,40 @@
+/*
```

Comment:
> 

@tejlmand, exactly how is RFC #40555 going to handle the problem that is addressed by the present PR. What is done here is overruling the definition that is given in the dts for the flash layout. More specific what is going to be the solution in the multi-image PR to support:
a. A non-mcuboot image,
b. A mcuboot image with only the slot0 image generated,
c. A mcuboot image with mcuboot + slot0 image generated,
(d. A mcuboot + slot1 image (encrypted) generated),

Reply:
> @tejlmand, exactly how is RFC #40555 going to handle the problem that is addressed by the present PR. What is done here is overruling the definition that is given in the dts for the flash layout. More specific what is going to be the solution in the multi-image PR to support:

For this PR, I only see a single `mcuboot`, so that must be `a` (being the case without this PR) and `b`.

Not `c` and `d`.

Regarding #40555, the topic you raise is outside the specific scope of that PR, but it opens up for a broader use-case, so let me sketch the general idea.

With #40555 it is possible to build the sample with or without mcuboot (or any other multi image, like the `openamp_remote` case).

In such a scenario, which would also be preparing for system devicetree, we can then apply a given set of overlay when a given bootloader is used.

If mcuboot is enabled when building, then we can ensure a devicetree setup with a flash layout suitable for mcuboot is selected and applied to all images part of the build.
This will further avoid the risk of the application being build with one `snippet` and `mcuboot` with another because you forgot to apply the same settings there.

The #40555 itself doesn't address this possibility, because it's just an RFC and I have tried to find the balance between amount of time spent to present the idea, while not yet spend too much time if the direction is rejected.